cpu/stm32f4/periph/gpio.c

```diff
@@ -25,598 +25,196 @@
-        return res;
+    /* enable clock */
+    RCC->AHB1ENR |= (1 << _port_num(pin));
+    /* configure pull register */
+    port->PUPDR &= ~(3 << (2 * pin_num));
+    port->PUPDR |= (pullup << (2 * pin_num));
+    /* set direction */
+    if (dir == GPIO_DIR_OUT) {
+        port->MODER &= ~(2 << (2 * pin_num));   /* set pin to output mode */
+        port->MODER |= (1 << (2 * pin_num));
+        port->OTYPER &= ~(1 << pin_num);        /* set to push-pull */
+        port->OSPEEDR |= (3 << (2 * pin_num));  /* set to high speed */
+        port->ODR &= ~(1 << pin_num);           /* set pin to low signal */
     }
+    else {
+        port->MODER &= ~(3 << (2 * pin_num));   /* configure pin as input */
+    }
+    return 0;
+}
 
-    pin = gpio_pin_map[dev];
+int gpio_init_exti(gpio_t pin,
+                   gpio_pp_t pullup, gpio_flank_t flank,
+                   gpio_cb_t cb, void *arg)
+{
+    int pin_num = _pin_num(pin);
+    int port_num = _port_num(pin);
 
+    /* configure and save exti configuration struct */
+    exti_chan[pin_num].cb = cb;
+    exti_chan[pin_num].arg = arg;
     /* enable the SYSCFG clock */
     RCC->APB2ENR |= RCC_APB2ENR_SYSCFGEN;
-
-    /* enable IRQ */
-    switch (dev) {
-#if GPIO_0_EN
-        case GPIO_0:
-            GPIO_0_EXTI_CFG();
-            break;
-#endif
-#if GPIO_1_EN
-        case GPIO_1:
-            GPIO_1_EXTI_CFG();
-            break;
-#endif
-#if GPIO_2_EN
-        case GPIO_2:
-            GPIO_2_EXTI_CFG();
-            break;
-#endif
-#if GPIO_3_EN
-        case GPIO_3:
-            GPIO_3_EXTI_CFG();
-            break;
-#endif
-#if GPIO_4_EN
-        case GPIO_4:
-            GPIO_4_EXTI_CFG();
-            break;
-#endif
-#if GPIO_5_EN
-        case GPIO_5:
-            GPIO_5_EXTI_CFG();
-            break;
-#endif
-#if GPIO_6_EN
-        case GPIO_6:
-            GPIO_6_EXTI_CFG();
-            break;
-#endif
-#if GPIO_7_EN
-        case GPIO_7:
-            GPIO_7_EXTI_CFG();
-            break;
-#endif
-#if GPIO_8_EN
-        case GPIO_8:
-            GPIO_8_EXTI_CFG();
-            break;
-#endif
-#if GPIO_9_EN
-        case GPIO_9:
-            GPIO_9_EXTI_CFG();
-            break;
-#endif
-#if GPIO_10_EN
-        case GPIO_10:
-            GPIO_10_EXTI_CFG();
-            break;
-#endif
-#if GPIO_11_EN
-        case GPIO_11:
-            GPIO_11_EXTI_CFG();
-            break;
-#endif
-#if GPIO_12_EN
-        case GPIO_12:
-            GPIO_12_EXTI_CFG();
-            break;
-#endif
-#if GPIO_13_EN
-        case GPIO_13:
-            GPIO_13_EXTI_CFG();
-            break;
-#endif
-#if GPIO_14_EN
-        case GPIO_14:
-            GPIO_14_EXTI_CFG();
-            break;
-#endif
-#if GPIO_15_EN
-        case GPIO_15:
-            GPIO_15_EXTI_CFG();
-            break;
-#endif
+    /* initialize pin as input */
+    gpio_init(pin, GPIO_DIR_IN, pullup);
+    /* enable global pin interrupt */
+    if (pin_num < 5) {
+        NVIC_EnableIRQ(EXTI0_IRQn + pin_num);
```

Comment:
It is not needed anymore, as this is done in `cortexm_init()` in `cpu/cortexm_common/cortexm_init.c`.